cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/AppFabricTestBase.java

```diff
@@ -539,21 +539,36 @@ protected HttpResponse addArtifact(Id.Artifact artifactId, InputSupplier<? exten
     return addArtifact(artifactId, artifactContents, parents, Collections.emptySet());
   }
 
-
   protected HttpResponse addArtifact(Id.Artifact artifactId, InputSupplier<? extends InputStream> artifactContents,
                                      Set<ArtifactRange> parents, Set<PluginClass> pluginClasses) throws Exception {
+    if (pluginClasses != null && !pluginClasses.isEmpty()) {
+      return addArtifactHelper(artifactId, artifactContents, parents, GSON.toJson(pluginClasses));
+    } else {
+      return addArtifactHelper(artifactId, artifactContents, parents, null);
+    }
+  }
+
+  /**
+   * This method is a helper method which accepts GSON serialized form of plugin classes for testing purpose only.
+   * It's usage is discouraged. Please consider using {@link #addArtifact(Id.Artifact, InputSupplier, Set, Set)}
+   */
+  protected HttpResponse addArtifactHelper(Id.Artifact artifactId,
```

Comment:
I would suggest testing the behavior using one of the existing test methods like `addPluginArtifact()` or something. This way we don't have to expose a new method that can cause confusion.